lib/core/elastic_connector_actions.rb

```diff
@@ -117,9 +118,36 @@ def client
         @client ||= Utility::EsClient.new
       end
 
+      # should only be used in CLI
+      def ensure_content_index_exists(index_name, use_icu_locale = false, language_code = nil)
+        settings = Utility::Elasticsearch::Index::TextAnalysisSettings.new(:language_code => language_code, :analysis_icu => use_icu_locale).to_h
+        mappings = Utility::Elasticsearch::Index::Mappings.default_text_fields_mappings(:connectors_index => true)
+
+        body_payload = { settings: settings, mappings: mappings }
+        ensure_index_exists(index_name, body_payload)
+      end
+
       # should only be used in CLI
       def ensure_index_exists(index_name, body = {})
-        client.indices.create(:index => index_name, :body => body) unless client.indices.exists?(:index => index_name)
+        if client.indices.exists?(:index => index_name)
+          Utility::Logger.info("Index #{index_name} already exists. Checking mappings...")
+          response = client.indices.get_mapping(:index => index_name)
+          existing = response[index_name]['mappings']
+          Utility::Logger.info("New settings/mappings: #{body}")
+          if existing.empty?
+            Utility::Logger.info("Index #{index_name} has no mappings. Closing to create settings and mappings...")
+            client.indices.close(:index => index_name)
+            client.indices.put_settings(:index => index_name, :body => body[:settings], :expand_wildcards => 'all')
```

Comment:
this is updating `settings` if `mappings` is empty.

I'm not sure, is it the best practice to close the index before updating `settings`/`mappings`?

Reply:
It's impossible to do it otherwise. The operation will fail if the index is open.